src/script/components/asset/controls/audioSeekBar.ts

```diff
@@ -16,41 +16,46 @@
  * along with this program. If not, see http://www.gnu.org/licenses/.
  *
  */
-
+import ko from 'knockout';
 import {debounce} from 'underscore';
 
 import {interpolate} from 'Util/ArrayUtil';
 import {clamp} from 'Util/NumberUtil';
 
+interface Params {
+  src: HTMLAudioElement;
+  disabled: ko.Subscribable<boolean>;
+
+  // TODO: replace with proper Type once they are defined
+  asset: {meta: {loudness: number[]}};
+}
+
 class AudioSeekBarComponent {
-  /**
-   * Construct a audio seek bar that renders audio levels.
-   *
-   * @param {Object} params - Component parameters
-   * @param {HTMLElement} params.media_src - Media source
-   * @param {File} params.asset - Asset file
-   * @param {boolean} params.disabled - Disabled seek bar
-   * @param {Object} componentInfo - Component information
-   */
-  constructor(params, componentInfo) {
+  audioElement: HTMLAudioElement;
+  element: HTMLElement;
+  loudness: number[];
+  levels: HTMLSpanElement[];
+  _onResizeFired: () => void;
+
+  constructor(params: Params, {element}: ko.components.ComponentInfo) {
     this.dispose = this.dispose.bind(this);
     this.audioElement = params.src;
 
-    this.element = componentInfo.element;
+    this.element = element as HTMLElement;
     this.loudness = [];
     this.levels = [];
 
-    this.disabled = ko.computed(
+    ko.computed(
       () => {
         if (typeof params.disabled === 'function') {
           this.element.classList.toggle('element-disabled', params.disabled());
         }
       },
-      {disposeWhenNodeIsRemoved: componentInfo.element},
+      {disposeWhenNodeIsRemoved: element},
     );
 
     const assetMeta = params.asset.meta;
-    if (assetMeta !== null && assetMeta.loudness !== null) {
+    if (assetMeta?.loudness !== null) {
```

Comment:
Before we checked only for `null` and not for `undefined`, does this have any implications?

Reply:
It does not. It will still not work for null and undefined will have errored before.